src/controller/CHIPDeviceController.cpp

```diff
@@ -94,7 +94,7 @@ DeviceController::DeviceController()
 }
 
 CHIP_ERROR DeviceController::Init(NodeId localDeviceId, PersistentStorageDelegate * storageDelegate, System::Layer * systemLayer,
-                                  Inet::InetLayer * inetLayer)
+                                  Inet::InetLayer * inetLayer, uint32_t bluetoothAdapterId)
```

Comment:
I believe we  are adding too many parameters to init - this turns into very coupled code.

Could we instead force a run of ConfigureBle outside devicecontroller init and let that control the adapter id?

Reply:
Added a PreConfigureBle api for this, if it is not called before Init the Init function will call it.

Reply:
My complaint is that Controller has too much responsibility/functionality. I am questioning why the API to configure BLE is within the controller. I would rather expect it to be decoupled.

Controller can error out if the required subsystems are not configured correctly (e.g. cannot listen on various ports) however I find an extra indirection for BLE configuration to go through a controller to be odd.

Reply:
Moved the BLE init out of the controller.